apps/src/flowlab/views/flow/nodes/AiNode.tsx

```diff
@@ -0,0 +1,124 @@
+import FontAwesomeV6Icon from '@code-dot-org/component-library/fontAwesomeV6Icon';
+import {
+  Position,
+  Handle,
+  useReactFlow,
+  useNodeConnections,
+  useNodesData,
+  type Node,
+  type NodeProps,
+} from '@xyflow/react';
+import React, {memo, useEffect, useState, useCallback} from 'react';
+
+import askAi from '../../../flow/askAi';
+import {isTextNode, type MyNode} from '../../../flow/initialElements';
+
+// This node asks the aichat service a question entered in its input field, and
+// sends the response to its output handle as text.  It acepts optional context
+// via any text sent to its input handle.  It makes a new request whenever
+// shift-enter is pressed while its input has focus, and whenever its input
+// context changes.
+
+function AiNode({
+  id,
+  data,
+}: NodeProps<Node<{fieldText: string; askedText: string; text: string}>>) {
```

Comment:
minor: `currentFieldText`, `lastFieldText`, and `output` or similar?

Reply:
I will ponder this.  We want to differentiate between what's currently typed into the field that's potentially "unsaved/unasked" versus the last thing to have been "saved/asked".  (The asterisk will show when these are different to each other.)